File: leetcode/104_maximumdepthofbinarytree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def getDepth(self, root, depth):
        if root is None: 
            return depth
        leftD = self.getDepth(root.left, depth+1)
        rightD = self.getDepth(root.right, depth+1)
        return max(leftD, rightD)
    
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        return self.getDepth(root, 0)

File: leetcode/104_maximumdepthofbinarytree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function getDepth(root: TreeNode | null, currentDepth) {
    if (!root) {
        return currentDepth
    }

    const leftDepth = getDepth(root.left, currentDepth+1)
    const rightDepth = getDepth(root.right, currentDepth+1)
    return Math.max(leftDepth, rightDepth)
}

function maxDepth(root: TreeNode | null): number {
    return getDepth(root, 0)
};
